cmd/gardener-scheduler/app/gardener_scheduler.go

```diff
@@ -111,175 +81,117 @@ func NewCommandStartGardenerScheduler() *cobra.Command {
 			if err := opts.validate(args); err != nil {
 				return err
 			}
-			return opts.run(cmd.Context())
+			return runCommand(cmd, opts)
 		},
 	}
 
 	flags := cmd.Flags()
 	verflag.AddFlags(flags)
 	opts.AddFlags(flags)
+
 	return cmd
 }
 
-// GardenerScheduler represents all the parameters required to start the
-// Gardener scheduler.
-type GardenerScheduler struct {
-	Config            *config.SchedulerConfiguration
-	Identity          *gardencorev1beta1.Gardener
-	GardenerNamespace string
-	ClientMap         clientmap.ClientMap
-	Logger            *logrus.Logger
-	Recorder          record.EventRecorder
-	LeaderElection    *leaderelection.LeaderElectionConfig
-}
+func runCommand(cmd *cobra.Command, opts *Options) error {
+	// Load config file
+	config, err := configloader.LoadFromFile(opts.ConfigFile)
+	if err != nil {
+		return err
+	}
 
-// NewGardenerScheduler is the main entry point of instantiating a new Gardener Scheduler.
-func NewGardenerScheduler(ctx context.Context, cfg *config.SchedulerConfiguration) (*GardenerScheduler, error) {
-	// validate the configuration
-	if err := validation.ValidateConfiguration(cfg); err != nil {
-		return nil, err
+	// Validate the configuration
+	if err := validation.ValidateConfiguration(config); err != nil {
+		return err
 	}
 
+	// Add feature flags
+	if err := schedulerfeatures.FeatureGate.SetFromMap(config.FeatureGates); err != nil {
+		return err
+	}
+	kubernetes.UseCachedRuntimeClients = schedulerfeatures.FeatureGate.Enabled(features.CachedRuntimeClients)
```

Comment:
Probably we don't need this at all anymore for the `gardener-scheduler` now that it's a native ctrl-runtime component?

Reply:
You mean remove the `FeaturesGates` field entirely, even from the type?

Reply:
Yes, let's remove it from here: https://github.com/gardener/gardener/blob/master/pkg/scheduler/features/features.go